x/bank/keeper/grpc_query.go

```diff
@@ -219,18 +219,14 @@ func (k BaseKeeper) DenomMetadataByQueryString(ctx context.Context, req *types.Q
 		return nil, status.Errorf(codes.InvalidArgument, "empty request")
 	}
 
-	if err := sdk.ValidateDenom(req.Denom); err != nil {
-		return nil, status.Error(codes.InvalidArgument, err.Error())
-	}
-
-	metadata, found := k.GetDenomMetaData(ctx, req.Denom)
-	if !found {
-		return nil, status.Errorf(codes.NotFound, "client metadata for denom %s", req.Denom)
+	res, err := k.DenomMetadata(ctx, &types.QueryDenomMetadataRequest{
```

Comment:
why not call `DenomOwnersByQuery` here instead?

Reply:
`DenomOwnersByQuery` and `DenomMetadata` both are different